client/rpc/src/eth_pubsub.rs

```diff
@@ -250,250 +216,221 @@ where
+									let current_block =
+										UniqueSaturatedInto::<u64>::unique_saturated_into(
+											client.info().best_number,
+										);
 
-								PubSubSyncStatus::Detailed(SyncStatusMetadata {
-									syncing: true,
-									starting_block,
-									current_block,
-									highest_block,
-								})
-							} else {
-								PubSubSyncStatus::Simple(false)
+									PubSubSyncStatus::Detailed(SyncStatusMetadata {
+										syncing: true,
+										starting_block,
+										current_block,
+										highest_block,
+									})
+								} else {
+									PubSubSyncStatus::Simple(false)
+								}
 							}
-						}
-						// On connection subscriber expects a value.
-						// Because import notifications are only emitted when the node is synced or
-						// in case of reorg, the first event is emited right away.
-						let _ = sink
-							.feed(Ok(PubSubResult::SyncState(
-								status(Arc::clone(&client), Arc::clone(&network), starting_block)
-									.await,
-							)))
-							.await;
 
-						// When the node is not under a major syncing (i.e. from genesis), react
-						// normally to import notifications.
-						//
-						// Only send new notifications down the pipe when the syncing status changed.
-						client
-							.import_notification_stream()
-							.fold(
-								network.is_major_syncing(),
-								move |mut last_syncing_status, _| {
-									let client = Arc::clone(&client);
-									let network = Arc::clone(&network);
-									let mut sink = sink.clone();
-									async move {
-										let syncing_status = network.is_major_syncing();
-										if last_syncing_status != syncing_status {
-											last_syncing_status = syncing_status;
-											let _ = sink
-												.feed(Ok(PubSubResult::SyncState(
-													status(client, network, starting_block).await,
-												)))
-												.await;
-										}
-										last_syncing_status
-									}
-								},
-							)
-							.map(|_| ())
-							.await
-					}
-				});
+							// On connection subscriber expects a value.
+							// Because import notifications are only emitted when the node is synced or
+							// in case of reorg, the first event is emited right away.
+							let _ = sink
+ 									.send(&PubSubResult::SyncState(
+ 										status(client.clone(), network.clone(), starting_block.clone()).await,
+ 									));
+
+							// When the node is not under a major syncing (i.e. from genesis), react
+							// normally to import notifications.
+							//
+							// Only send new notifications down the pipe when the syncing status changed.
+							use std::sync::atomic::{AtomicBool, Ordering};
+							let last_syncing_status = AtomicBool::new(Arc::clone(&network).is_major_syncing());
+							let stream = client.clone().import_notification_stream().filter_map(|_| async {
+								let syncing_status = network.is_major_syncing();
+								if syncing_status
+ 									!= last_syncing_status.swap(syncing_status, Ordering::SeqCst)
```

Comment:
Why is this here again?

Reply:
Because we merged the same approach in the previous locked.

Reply:
Didn't we alter it to get rid of atomics?

Reply:
Let's see final jsonrpsee integration from frontier team or later will propose our implementation.